pkg/metrics/service-monitor.go

```diff
@@ -28,9 +28,11 @@ import (
 
 var ErrServiceMonitorNotPresent = fmt.Errorf("no ServiceMonitor registered with the API")
 
+type ServiceMonitorUpdater func(*monitoringv1.ServiceMonitor) error
+
 // CreateServiceMonitors creates ServiceMonitors objects based on an array of Service objects.
 // If CR ServiceMonitor is not registered in the Cluster it will not attempt at creating resources.
-func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Service) ([]*monitoringv1.ServiceMonitor, error) {
+func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Service, updaters... ServiceMonitorUpdater) ([]*monitoringv1.ServiceMonitor, error) {
```

Comment:
Nit: this does not follow `go fmt`'s suggested format, which is to put the ellipses on the type, not the variable name.

```suggestion
func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Service, updaters ...ServiceMonitorUpdater) ([]*monitoringv1.ServiceMonitor, error) {
```

As a side note, we should probably run `go fmt ./...` right after `go vet ./...` in the `hack/tests/sanity-check.sh` script, so that the `git diff` catches these in the future.

I'm good with adding that directly in this PR, if you don't mind adding it.

Reply:
:+1: